assets/js/timetree.js

```diff
@@ -512,11 +512,26 @@ function TreeGraph({ nodes, links, centuries }) {
 
   const panel = document.querySelector("#panel");
   d3.select("aside .close").on("click", function () {
+    // Close panel and deselect
+    close_panel(panel);
+  });
+
+  // Close panel function
+  function close_panel(panel) {
     panel.parentElement.classList.remove("show-panel");
     panel.parentElement.classList.add("closed");
     Leaf.deselectAll();
     Leaf.closeLeafDetails();
-  });
+  }
+
+  // Also allow Escape key to close window
+  document.onkeydown = function (evt) {
+    evt = evt || window.event;
+    var isEscape = evt.key === "Escape" || evt.key === "Esc";
+    if (isEscape) {
+      close_panel(panel);
+    }
```

Comment:
This is great, I love using it already. Would it make sense to use a switch statement like in the MDN example so it's easier to add new events? https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key 

I've made the leaves focusable so you can tab through the tree, but right now you can't activate them (for keyboard accessibility, I think we'll need to make them behave more like buttons). IDK if we'll put that logic here, though.

Reply:
Great idea!